open-weather-connector-demo/src/com/axonivy/connector/openweather/managedbean/ForecastWeatherBean.java

```diff
@@ -0,0 +1,431 @@
+package com.axonivy.connector.openweather.managedbean;
+
+import java.io.Serializable;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import javax.annotation.PostConstruct;
+import javax.faces.bean.ManagedBean;
+import javax.faces.bean.ViewScoped;
+import javax.faces.context.FacesContext;
+
+import org.apache.commons.lang.StringUtils;
+import org.openweathermap.api.data2_5.client.Forecast;
+import org.openweathermap.api.data2_5.client.WeatherRecord;
+import org.primefaces.model.charts.ChartData;
+import org.primefaces.model.charts.bar.BarChartDataSet;
+import org.primefaces.model.charts.bar.BarChartModel;
+import org.primefaces.model.charts.bar.BarChartOptions;
+import org.primefaces.model.charts.line.LineChartDataSet;
+import org.primefaces.model.charts.line.LineChartModel;
+import org.primefaces.model.charts.line.LineChartOptions;
+
+import com.axonivy.connector.openweather.dto.DailyForecast;
+import com.axonivy.connector.openweather.dto.DailyForecastDisplayInfo;
+import com.axonivy.connector.openweather.service.ForecastService;
+import com.axonivy.connector.openweather.util.Constants;
+import com.axonivy.connector.openweather.util.DateTimeFormatterUtilities;
+
+@ManagedBean
+@ViewScoped
+public class ForecastWeatherBean implements Serializable {
+
+	private static final long serialVersionUID = 4700043015312367231L;
+
+	private String typeOfDegree;
+	private LocalDate selectedDate;
+	private LocalTime selectedTime;
+	private int selectedDateIndex;
+	private int selectedTimeIndex;
+	private String cityName;
+	private String time12HourName;
+	private String dateName;
+	private ZoneId zoneId;
+	private String speedUnit;
+	private String units;
+
+	private int currentTemperature;
+	private int currentHumidity;
+	private float currentWindSpeed;
+	private String currentWeatherIconCode;
+	private String currentWeatherDetail;
+
+	private List<WeatherRecord> threeHourlyFiveDayForecasts;
+	private List<DailyForecast> dailyForecasts;
+	private List<DailyForecastDisplayInfo> dailyForecastDisplayInfos;
+
+	private String searchCityName;
+	private String searchCountryCode;
+	private String searchStateCode;
+
+	private LineChartModel temperatureModel;
+	private String temperatureData;
+	private BarChartModel precipitationModel;
+	private int chartWindowSize;
+	private int currentChartWindowStartX;
+	private int currentChartWindowEndX;
+
+	@PostConstruct
+	public void init() {
+		searchCityName = Constants.DEFAULT_SEARCH_CITY_NAME;
```

Comment:
Should we declare the default values in the variables.yaml file and users can change them as they wish?

Reply:
That's a good idea, thanks!